addons/website_sale/controllers/main.py

```diff
@@ -1271,20 +1196,11 @@ def shop_address_submit(
 
         order_sudo._update_address(partner_sudo.id, partner_fnames)
 
-        if is_anonymous_cart:
+        if is_anonymous_customer:
```

Comment:
Anonymous cart and anonymous customer are two different logics.
The user could be anonymous when the cart isn't.
This code only makes sense when the cart was assigned to the Public Partner, not when the current user is the public user but he already provided his address for the cart.